f5_openstack_agent/lbaasv2/drivers/bigip/icontrol_driver.py

```diff
@@ -1348,6 +1369,8 @@ def _update_listener_status(self, service):
                             plugin_const.ACTIVE,
                             listener['operating_status']
                         )
+                        listener['provisioning_status'] = \
```

Comment:
This looks incorrect.  The previous rpc call sets the pool status to that of the listener.  Your new code forces listener status to active (which may differ from what was just sent over rpc).